x-pack/filebeat/module/envoyproxy/README.md

```diff
@@ -0,0 +1,128 @@
+
+## Deployment Scenario #1: envoy native deployment
+
+Make sure to update envoy configuration to enable log plugin. This module assumes that envoy log
+entries will be written to /var/log/envoy.log. Should it be not the case, please point the module 
+log path to the path of the log file. 
+
+Update filebeat.yml to point to Elasticsearch and Kibana. 
+Setup Filebeat.
+```
+./filebeat setup --modules envoyproxy -e
+```
+
+Enable the Filebeat envoyproxy module
+```
+./filebeat modules enable envoyproxy
+```
+
+Start Filebeat
+```
+./filebeat -e
+```
+
+Now, the Envoy logs and dashboard should appear in Kibana.
+
+
+## Deployment Scenario #2: envoy for kubernetes 
+
+For Kubernetes deployment, the filebeat daemon-set yaml file needs to be deployed to the 
+Kubernetes cluster. Sample configuration files is provided under the `beats/deploy/filebeat` 
+directory, and can be deployed by doing the following:
+```
+kubectl apply -f filebeat
+```
+
+#### Note the following section in the ConfigMap, make changes to the yaml file if necessary
+```
+  filebeat.autodiscover:
+    providers:
+      - type: kubernetes
+        hints.enabled: true
+        default.disable: true
+
+  processors:
+    - add_kubernetes_metadata:
+        in_cluster: true
+```
+
+This enables auto-discovery and hints for filebeat. When default.disable is set to true (default value is false), it will disable log harvesting for the pod/container, unless it has specific annotations enabled. This gives users more granular control on kubernetes log ingestion. The `add_kubernetes_metadata` processor will add enrichment data for Kubernetes to the ingest logs.
+
+#### Note the following section in the DaemonSet, make changes to the yaml file if necessary
+```
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: filebeat
+  namespace: kube-system
+  labels:
+    k8s-app: filebeat
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: filebeat
+    spec:
+      serviceAccountName: filebeat
+      terminationGracePeriodSeconds: 30
+      containers:
+      - name: filebeat
+        image: docker.elastic.co/beats/filebeat:%VERSION%
+        args: [
+          "sh", "-c", "filebeat setup -e --modules envoyproxy -c /etc/filebeat.yml && filebeat -e -c /etc/filebeat.yml"
+        ]
+        env:
+        # Edit the following values to reflect your setup accordingly
+        - name: ELASTICSEARCH_HOST
+          value: 192.168.99.1
+        - name: ELASTICSEARCH_USERNAME
+          value: elastic
+        - name: ELASTICSEARCH_PASSWORD
+          value: changeme
+        - name: KIBANA_HOST
+          value: 192.168.99.1
+```
+
+The module setup step can also be done separately without Kubernetes if applicable, and in that case, the args can be simplified to:
+```
+        args: [
+          "sh", "-c", "filebeat -e -c /etc/filebeat.yml"
+        ]
+```
+
+#### Sample Deployment for envoy, using ambassador as an example. Note the annotations.
```

Comment:
I'm not following here.

Ambassador is an API Gateway. If this is intended to explain a working environment with Envoy, some services should configured at Ambassador, ... but imho, adding Ambassador might be overcomplicating the scenario.

What about deploying Envoy + 2048 (you can't go wrong with 2048!), and configuring Envoy to expose it?
https://github.com/kubernetes-sigs/aws-alb-ingress-controller/blob/master/docs/examples/2048/2048-deployment.yaml

In fact, there is no need to configure anything, you can just run Envoy and I'm pretty sure it will spit some logs

Reply:
This is for the common k8s deployment scenario where Ambassador is used. Users can choose any others, but it is the same to the module since the configuration is done via k8s annotation.